src/core/tests/pattern.cpp

```diff
@@ -508,6 +510,68 @@ TEST(pattern, matching_optional) {
                          std::make_shared<op::v0::Abs>(c)));
 }
 
+TEST(pattern, optional_full_match) {
+    Shape shape{};
+    auto model_input1 = std::make_shared<op::v0::Parameter>(element::i32, shape);
+    auto model_input2 = std::make_shared<op::v0::Parameter>(element::i32, shape);
+    auto model_add = std::make_shared<op::v1::Add>(model_input1->output(0), model_input2->output(0));
+    auto model_relu = std::make_shared<op::v0::Relu>(model_add->output(0));
+
+    auto pattern_add = ov::pass::pattern::optional<op::v1::Add>();
+    auto pattern_relu = std::make_shared<op::v0::Relu>(pattern_add->output(0));
+
+    TestMatcher tm;
+
+    ASSERT_TRUE(tm.match(pattern_relu, model_relu));
```

Comment:
I would recommend to test pattern_map in these tests
Please check that pattern_map contains pattern_add/pattern_relu keys and the corresponding values are equal to model_add and model_relu.

We can do it in the next PR, not a problem.